cmake/Jucer2CMake/main.cpp

```diff
@@ -0,0 +1,302 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "JuceHeader.h"
+
+#include <jucer_PresetIDs.h>
+#include <iostream>
+#include <string>
+#include <vector>
+#include <tuple>
+#include <map>
+#include <list>
+#include <fstream>
+
+void printError(const std::string &error) {
+  std::cerr << "error: " << error << std::endl;
+}
+
+String makeValidIdentifier(String s, bool capitalise, bool removeColons,
+                           bool allowTemplates) {
+  if (s.isEmpty())
+    return "unknown";
+
+  if (removeColons)
+    s = s.replaceCharacters(".,;:/@", "______");
+  else
+    s = s.replaceCharacters(".,;/@", "_____");
+
+  for (int i = s.length(); --i > 0;)
+    if (CharacterFunctions::isLetter(s[i]) &&
+        CharacterFunctions::isLetter(s[i - 1]) &&
+        CharacterFunctions::isUpperCase(s[i]) &&
+        !CharacterFunctions::isUpperCase(s[i - 1]))
+      s = s.substring(0, i) + " " + s.substring(i);
+
+  String allowedChars(
+      "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_ 0123456789");
+  if (allowTemplates)
+    allowedChars += "<>";
+
+  if (!removeColons)
+    allowedChars += ":";
+
+  StringArray words;
+  words.addTokens(s.retainCharacters(allowedChars), false);
+  words.trim();
+
+  String n(words[0]);
+
+  if (capitalise)
+    n = n.toLowerCase();
+
+  for (int i = 1; i < words.size(); ++i) {
+    if (capitalise && words[i].length() > 1)
+      n << words[i].substring(0, 1).toUpperCase()
+        << words[i].substring(1).toLowerCase();
+    else
+      n << words[i];
+  }
+
+  if (CharacterFunctions::isDigit(n[0]))
+    n = "_" + n;
+
+  if (CPlusPlusCodeTokeniser::isReservedKeyword(n))
+    n << '_';
+
+  return n;
+}
+
+int main(int argc, char *argv[]) {
+  // Test initial conditions
+  if (argc != 3) {
+    std::cerr << "usage: Jucer2CMake <jucer_project_file> "
+                 "<reproducer_library_path>"
+              << std::endl;
+    return 1;
+  }
+
+  // Parse command line arguments
+  std::vector<std::string> args{argv, argv + argc};
+  const auto &jucerFilePath = args.at(1);
+  const auto &reproducerLibraryPath = args.at(2);
+  const auto currentWorkingDirectory =
+      juce::File::getCurrentWorkingDirectory().getFullPathName().toStdString();
+  const auto cmakeProjectJucerPrefix =
+      makeValidIdentifier(
+          juce::File(jucerFilePath).getFileNameWithoutExtension() + "_jucer",
+          false, true, false)
```

Comment:
@MartyLake: instead of passing `false, true, false`, let's change the function to not take these parameters.